contrib/list_missing_pkg_tags.jl

```diff
@@ -0,0 +1,89 @@
+using LibGit2
+
+const JULIA_REPO_URL = "https://github.com/JuliaLang/julia.git"
+const JULIA_REPO_DIR = "julia"
+const PKG_VERSION_PATH = "stdlib/Pkg.version"
+const PKG_REPO_URL = "https://github.com/JuliaLang/Pkg.jl.git"
+const PKG_REPO_DIR = "Pkg.jl"
+
+function checkout_or_update_repo(url, dir)
+    if isdir(dir)
+        println("Updating existing repository: $dir")
+        repo = LibGit2.GitRepo(dir)
+        LibGit2.fetch(repo)
+    else
+        println("Cloning repository: $url")
+        LibGit2.clone(url, dir)
+    end
+end
+
+function get_tags(repo)
+    refs = LibGit2.ref_list(repo)
+    tags = filter(ref -> startswith(ref, "refs/tags/"), refs)
+    return sort!(replace.(tags, "refs/tags/" => ""))
+end
+
+function is_stable_v1_release(tag)
+    return occursin(r"^v\d+\.\d+\.\d+$", tag) && VersionNumber(tag) >= v"1.0.0"
+end
+
+function extract_pkg_sha1(text::AbstractString)
+    m = match(r"PKG_SHA1\s*=\s*([a-f0-9]{40})", text)
+    return m !== nothing ? m[1] : nothing
+end
+
+function get_commit_hash_for_pkg_version(repo, tag)
+    try
+        tag_ref = LibGit2.GitReference(repo, "refs/tags/" * tag)
+        LibGit2.checkout!(repo, string(LibGit2.GitHash(LibGit2.peel(tag_ref))))
+        version_file = joinpath(JULIA_REPO_DIR, PKG_VERSION_PATH)
+        if isfile(version_file)
+            return extract_pkg_sha1(readchomp(version_file))
+        else
+            println("Warning: Pkg.version file missing for tag $tag")
+            return nothing
+        end
+    catch e
+        println("Error processing tag $tag: ", e)
```

Comment:
What's the case where we might hit this error? Do we need to catch it? Could we just rethrow it instead?